src/main/scala/com/fulcrumgenomics/bam/EstimatePoolingFractions.scala

```diff
@@ -163,20 +169,18 @@ class EstimatePoolingFractions
   }
 
   /** Generates an iterator over non-filtered bi-allelic SNPs where all the required samples are genotyped. */
-  def constructVcfIterator(in: VCFFileReader, intervals: Option[IntervalList], samples: Array[String]): Iterator[VariantContext] = {
-    val vcfIterator: Iterator[VariantContext] = intervals match {
+  def constructVcfIterator(in: VcfSource, intervals: Option[IntervalList], samples: Seq[String]): Iterator[Variant] = {
+    val iterator: Iterator[Variant] = intervals match {
       case None     => in.iterator
-      case Some(is) => ByIntervalListVariantContextIterator(in, is)
+      case Some(is) => is.flatMap(i => in.query(i.getContig, i.getStart, i.getEnd))
     }
 
-    val samplesAsUtilSet = CollectionUtil.makeSet(samples:_*)
-
-    vcfIterator
-      .filterNot(_.isFiltered)
-      .map(_.subContextFromSamples(samplesAsUtilSet, true))
-      .filter(v => v.isSNP && v.isBiallelic && !v.isMonomorphicInSamples)
-      .filter(_.getNoCallCount == 0)
-      .filter(v => v.getGenotypesOrderedByName.iterator.forall(gt => gt.getGQ >= this.minGenotypeQuality))
+    iterator
+      .filter(v => v.filters.isEmpty || v.filters == Variant.PassingFilters)
+      .filter(v => v.alleles.size == 2 && v.alleles.forall(a => a.value.length == 1))  // Just biallelic SNPs
+      .filter(v => samples.map(v.gt).forall(gt => gt.isFullyCalled && (this.minGenotypeQuality <= 0 || gt.get[Int]("GQ").exists(_ >= this.minGenotypeQuality))))
+      .map   (v => v.copy(genotypes=v.genotypes.filter { case (s, _) => samples.contains(s)} ))
```

Comment:
Should we subset to the samples we care about before we apply the genotype-quality and fully called filters above?

Reply:
It would make the code a little shorter, but I was trying to avoid a `copy()` operation on variants that we're not going to use.  Hence above I check only the genotypes for samples that we care about, and then do the copy.